api/src/main/java/org/jboss/forge/furnace/container/simple/Producer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2014 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Eclipse Public License version 1.0, available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.jboss.forge.furnace.container.simple;
+
+/**
+ * Interface used to register a Service to be a producer.
+ * 
+ * <p>
+ * To register a {@link Producer}, register it as a {@link Service} or {@link SingletonService}.
+ * <p>
+ * 
+ * <pre>
+ * public class ExampleService extends ProducerService<MyObject>
+ * {
+ *    &#64;Override
+ *    public MyObject get()
+ *    {
+ *       // r MyObject
+ *    }
+ * }
+ * </pre>
+ * <p>
+ * Example registration file:
+ * </p>
+ * 
+ * <pre>
+ * META-INF/services/org.jboss.forge.furnace.container.simple.Service
+ * -------
+ * org.example.ExampleService
+ * org.example.ExampleService2
+ * org.my.custom.MyService
+ * -------
+ * </pre>
+ * 
```

Comment:
<b>WARNING: Producer services must NEVER be directly accessed outside of the Addon in which they are defined. This will result in compilation failures in JDK8+.</b>